### PR TITLE
Feature/783 verify kafka topic access UI

### PR DESCRIPTION
**Pull request by jozefbakus**
Implemented frontend for: #783

### CODE REVIEW

File: ui/src/app/components/workflows/workflow-form/jobs/job/job-types/hyperdrive-job/hyperdrive-job.component.ts [hunk 1]
```diff
@@ -170,7 +173,23 @@ export class HyperdriveJobComponent implements OnInit, OnDestroy {
     this.jobParametersChange.emit({ ...this.jobParameters, additionalSparkConfig: additionalSparkConfig });
   }
 
+  verifyKafkaTopicAuthorizations = (topic: string): void => {
+    !!this.getKafkaTopicAuthorizationsSubscription && this.getKafkaTopicAuthorizationsSubscription.unsubscribe();
```
Comment (kevinwallimann) [inline ui/src/app/components/workflows/workflow-form/jobs/job/job-types/hyperdrive-job/hyperdrive-job.component.ts]: Why do you unsubscribe here?

File: ui/src/app/components/workflows/workflow-form/jobs/job/job-types/hyperdrive-job/hyperdrive-job.component.ts [hunk 2]
```diff
@@ -170,7 +173,23 @@ export class HyperdriveJobComponent implements OnInit, OnDestroy {
...
+    this.getKafkaTopicAuthorizationsSubscription = this.confluentService.getKafkaTopicAuthorizations(topic).subscribe((kafkaTopicAuth) => {
+      if (kafkaTopicAuth.exists) {
+        this.toastrService.success(
+          `Topic exists. Read access: ${kafkaTopicAuth.hasReadAccess ? 'Yes' : 'No'}. Write access: ${
+            kafkaTopicAuth.hasWriteAccess ? 'Yes' : 'No'
+          }.`,
+        );
+      } else {
+        this.toastrService.error('Topic does not exist or we do not have access to it!');
```
Comment (kevinwallimann) [inline ui/src/app/components/workflows/workflow-form/jobs/job/job-types/hyperdrive-job/hyperdrive-job.component.ts]: `we do not have access` -> `the hyperdrive service-user has no access`

**Review by kevinwallimann — COMMENTED** [review]
I think the UX of this button is ok, but a little rough. If the user loads the workflow, it would make sense that there would already be a checkmark indicating that the topic exists and read / write access is granted.
If the user creates a workflow, it would probably make sense to store the response of the topic verification and again indicate next to the "Verify Topic" text with a checkmark that the read access is granted (for the source topic), or the write access is granted (for the destination topic)

File: ui/src/app/components/workflows/workflow-form/jobs/job/job-types/hyperdrive-job/hyperdrive-job.component.ts (hunk 2 above)
Comment (jozefbakus) [reply ui/src/app/components/workflows/workflow-form/jobs/job/job-types/hyperdrive-job/hyperdrive-job.component.ts]: Done

File: ui/src/app/components/workflows/workflow-form/jobs/job/job-types/hyperdrive-job/hyperdrive-job.component.ts (hunk 1 above)
Comment (jozefbakus) [reply ui/src/app/components/workflows/workflow-form/jobs/job/job-types/hyperdrive-job/hyperdrive-job.component.ts]: To cancel previous call

**Comment by jozefbakus** [comment]
> I think the UX of this button is ok, but a little rough. If the user loads the workflow, it would make sense that there would already be a checkmark indicating that the topic exists and read / write access is granted. If the user creates a workflow, it would probably make sense to store the response of the topic verification and again indicate next to the "Verify Topic" text with a checkmark that the read access is granted (for the source topic), or the write access is granted (for the destination topic)

I changed it a bit. I don't want to store any response. 
In edit mode I left it as it was.
In show mode on page load it also pulls up to date status and shows it on the page next to the topic name

File: ui/src/app/components/workflows/workflow-form/jobs/job/job-types/hyperdrive-job/hyperdrive-job.component.html [hunk 3]
```diff
@@ -110,14 +116,16 @@
     </app-string-part>
   </div>
   <div *ngSwitchCase="hyperdriveTypes.OFFLOAD_PUBLISH_TOPIC">
-    <app-string-part
+    <app-string-with-verify-part
       [isShow]="isShow"
       [name]="'Source topic'"
+      [verifyButtonText]="'Verify topic'"
+      [verifyCall]="verifyKafkaTopicAuthorizations(true)"
```
Comment (kevinwallimann) [inline ui/src/app/components/workflows/workflow-form/jobs/job/job-types/hyperdrive-job/hyperdrive-job.component.html]: Should be `false`, as discussed

File: ui/src/app/constants/texts.constants.ts [hunk 4]
```diff
@@ -153,4 +153,11 @@ export const texts = {
   LOAD_NOTIFICATION_RULE_FAILURE_NOTIFICATION: "Sorry, notification rule couldn't be loaded. Please try again.",
   LOAD_HISTORY_FOR_NOTIFICATION_RULE_FAILURE_NOTIFICATION: 'Sorry, something went wrong. Try again.',
   LOAD_NOTIFICATION_RULES_FROM_HISTORY_FAILURE_NOTIFICATION: 'Sorry, something went wrong. Try again.',
+
+  HYPERDRIVE_JOB_NO_TOPIC_ACCESS: 'Topic does not exist or the hyperdrive service-user has no access!',
+  HYPERDRIVE_JOB_NO_WRITE_ACCESS: 'No write access!',
+  HYPERDRIVE_JOB_WRITE_ACCESS: 'The user has write access!',
+  HYPERDRIVE_JOB_NO_READ_ACCESS: 'No read access!',
+  HYPERDRIVE_JOB_READ_ACCESS: 'The user has read access!',
+  HYPERDRIVE_JOB_COULD_NOT_VERIFY_ACCESS: 'Sorry, something went wrong. Try again.',
```
Comment (kevinwallimann) [inline ui/src/app/constants/texts.constants.ts]: `Try again` -> `Please try again later`

**Review by kevinwallimann — COMMENTED** [review]
Very nice!

File: ui/src/app/components/workflows/workflow-form/jobs/job/job-types/hyperdrive-job/hyperdrive-job.component.html (hunk 3 above)
Comment (jozefbakus) [reply ui/src/app/components/workflows/workflow-form/jobs/job/job-types/hyperdrive-job/hyperdrive-job.component.html]: Done

File: ui/src/app/constants/texts.constants.ts (hunk 4 above)
Comment (jozefbakus) [reply ui/src/app/constants/texts.constants.ts]: Done

**Review by kevinwallimann — APPROVED** [review]

**Pull request merged**